File: hackerrank/easy/0002-the-hurdle-race/swift/solution.swift
// solution.swift

func hurdleRace(k: Int, height: [Int]) -> Int {
    let maxHurdle = height.max() ?? 0
    return k >= maxHurdle ? 0 : maxHurdle - k
}

// --- demo run ---
let k = 4
let hurdles = [1, 6, 3, 5, 2]
print(hurdleRace(k: k, height: hurdles))  // 2

File: hackerrank/easy/0002-the-hurdle-race/python/solution.py
# solution.py

from typing import List

def hurdle_race(k: int, height: List[int]) -> int:
    max_hurdle = max(height) if height else 0
    return 0 if k >= max_hurdle else max_hurdle - k


# --- demo run ---
if __name__ == "__main__":
    k = 4
    hurdles = [1, 6, 3, 5, 2]
    print(hurdle_race(k, hurdles))  # 2